### ISSUE TITLE
Offline config test bot took 10 min to load public config

### ISSUE THREAD

**Issue by m4xson** [taniman/profit-trailer#2050]
## ProfitTrailer info
Version: 2.1.9

 
Exchange:
- [ ] Bittrex
- [x ] Binance
- [ ] Poloniex

 
Market:
- [x ] BTC
- [ ] ETH
- [ ] USDT
- [ ] BNB
- [ ] XMR

 
Operating System:
- [x ] Windows 10
- [ ] Ubuntu (please specify version)
- [ ] Other linux distro (please specify)
- [ ] Mac OS

## Expected Behavior
The bot should load an online config on startup and save it within a few seconds

## Current Behavior
I started a fresh test bot with 2.1.9. I put in the password etc and then I selected to load the public config Old Faithful V4

## Troubleshooting
Contacted support. This may be a known issue.

## Possible Solution
Use online configs until fixed.

## Steps to Reproduce

1. Set server.disable_online_config = true
2. Clear all logs and files appart from application.properties
3. Download PT 2.1.9 and copy files to application folder.
4. Run PT and enter password and default api key
5. Select to load public config Old Faithful V4
6. See the error message about:
2018-08-28 07:22:22 INFO ProfitTrailerService - Invalid market!
2018-08-28 07:22:22 INFO ProfitTrailerService - DCA Heartbeat
2018-08-28 07:22:23 INFO SettingsService - The requested config does not exist
2018-08-28 07:22:23 INFO ConfigApi - We could not find your config OldFaithfulV4
7. Go to the advanced settings and see just loading in there as per the screenshot.
![public config not loading](https://user-images.githubusercontent.com/42469965/44720047-4819ee80-aabd-11e8-8929-c985c736b077.JPG)
8. Wait 10 mins for it to start working as expected.

## Configuration

 Added.

## Log files/console output
2018-08-28 07:19:22 INFO Application - PT Version: 2.1.9
2018-08-28 07:19:22 INFO Application - Java Version: 1.8.0_181
2018-08-28 07:19:27 INFO Application - Please finish the setup process in your browser
2018-08-28 07:19:27 INFO Application$a - Server initialized with port: 8081
2018-08-28 07:22:21 INFO ProfitTrailerService - Cache Heartbeat
2018-08-28 07:22:22 INFO Util - Detected configuration changes
2018-08-28 07:22:22 INFO ProfitTrailerService - PAIRS Heartbeat
2018-08-28 07:22:22 INFO ProfitTrailerService - Invalid market!
2018-08-28 07:22:22 INFO ProfitTrailerService - DCA Heartbeat
2018-08-28 07:22:23 INFO SettingsService - The requested config does not exist
2018-08-28 07:22:23 INFO ConfigApi - We could not find your config OldFaithfulV4
2018-08-28 07:32:22 INFO ProfitTrailerService - Cache Heartbeat
2018-08-28 07:32:23 INFO ProfitTrailerService - PAIRS Heartbeat
2018-08-28 07:32:24 INFO ProfitTrailerService - DCA Heartbeat
[ProfitTrailer.log](https://github.com/taniman/profit-trailer/files/2327750/ProfitTrailer.log)
[OldFaithful4-DCA.txt](https://github.com/taniman/profit-trailer/files/2327753/OldFaithful4-DCA.txt)
[OldFaithful4-INDICATORS.txt](https://github.com/taniman/profit-trailer/files/2327754/OldFaithful4-INDICATORS.txt)
[OldFaithful4-PAIRS.txt](https://github.com/taniman/profit-trailer/files/2327755/OldFaithful4-PAIRS.txt)

**Comment by taniman** [comment]
I just tried, and took me less than 10 seconds to get up and running .

![image](https://user-images.githubusercontent.com/6751341/44720401-ffffc980-aac6-11e8-9343-fe709dd9922a.png)

I imagine the answer is yes, but did you try refreshing the page? The GUI tries to load the config too fast causing the invalid market message. (2.1.10 will have an improved version). But the copy of a public config will take as long as it needs. and actually there is a 1 minute timeout on api server. 
So either the copy worked and you needed to refresh the page or the copy did not work.

I am sorry there was a delay with loading config. But these are the steps roughly
1. We contact the server to get the config.
2. We copy the ocnfig
3. We save the config.
4. If there is connectivity issues ether from the API server or your machine, it may take a while to complete the request. 

I would not call this a bug cause there is nothing I can do from within the bot itself to make it faster. If a copy fails you try again. 

With that said. Once again 2.1.10 will have a copy that does not throw the invalid market message

**Comment by m4xson** [comment]
Did I refresh? I assumed the PT monitor page auto refreshed TBH. It just looked like it was stuck. If you look at the log it did take several minutes to load the config. Maybe a warning in the GUI about the time to load could avoid confusion. I was convinced the boy had stalled but it just wasn't handling the time to time to download the config that well. Hopefully 2.1.10 will handle this better like you say.

Thanks for looking into it anyway!

**Comment by taniman** [comment]
That's the thing. Your log does not say it took any amount of time to load the config. 

```
2018-08-28 07:22:21 INFO ProfitTrailerService - Cache Heartbeat
2018-08-28 07:22:22 INFO Util - Detected configuration changes
2018-08-28 07:22:22 INFO ProfitTrailerService - PAIRS Heartbeat
2018-08-28 07:22:22 INFO ProfitTrailerService - Invalid market!
2018-08-28 07:22:22 INFO ProfitTrailerService - DCA Heartbeat
2018-08-28 07:22:23 INFO SettingsService - The requested config does not exist
2018-08-28 07:22:23 INFO ConfigApi - We could not find your config OldFaithfulV4
```

There is 3 seconds between these lines. And at the 22 second mark it says detected configuration changes. That's the moment the new configs were loaded I presume. 
And after the GUI tried loading an invalid config and such so it looked like nothing good happened.

The GUI itself does not do an F5. It does do background refreshes, but that's not the same as an F5. 

In any case 2.1.10 will be a step in the right direction

**Comment by m4xson** [comment]
I think the 23 second mark where it says:
2018-08-28 07:22:23 INFO SettingsService - The requested config does not exist
2018-08-28 07:22:23 INFO ConfigApi - We could not find your config OldFaithfulV4
Was where I started clicking around in the config to see why the market was wrong and I might have clicked on the button to switch configs but there was only one there which wasn't loaded. This was when the advanced config just said 'Loading' and clicking around did nothing to change that.

I just loaded another public config and there was no apparent issue so this problem must be particular to a fresh bot in a new folder.
